### PR TITLE
Immune deconv

### PR DESCRIPTION
**Pull request by komalsrathi**
### Purpose/implementation Section

Immune Deconvolution

#### What scientific question is your analysis addressing?

Update with v15 data (removed redundancy in histology file)

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?

Yes

#### Reproducibility Checklist

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

#### Documentation Checklist

- [x] This analysis module has a `README` and it is up to date.
- [x] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

**Comment by jaclyn-taroni** [comment]
Hi @komalsrathi - this looks good to me and I am happy to approve and merge. I wanted to ask first if you wanted to use this PR as an opportunity to address your concerns about separating out the non-brain cancers from https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/519#issuecomment-586352879.

**Comment by komalsrathi** [comment]
@jaclyn-taroni I was just going to comment on that. I have not heard back from Payal - who initially opened the issue. I thought the original heatmaps were pretty reasonable (per the methods). I think it only makes sense to separate the brain tumors and non-brain ones if any interesting inference can be drawn from it. If we are keeping the original heatmaps (which I would prefer), then I don't think lymphomas should be kept out because even though it is an N of 1, we are just visualizing the average scores per histology per cell type and that 1 lymphoma sample shows nicely that it is enriched in B-cell types. @jharenza thoughts?

**Comment by jaclyn-taroni** [comment]
Tagging @jainpayal022 for discussion

**Comment by jharenza** [comment]
@komalsrathi and @jaclyn-taroni - I think our rationale for separating these was to determine if we could see any signal in CNS tumors, since the non-CNS tumors were so enriched that they blew out any signal. I do think histiocytic and lymphomas are really nice controls and worth keeping, even if lymphoma has an N of 1. Separating these out, we now see some signal in germ cell, mesenchymal, LGAT, metastatic, some lower level of others. I think how we go forward depends on what we would like to say about the tumors - if we put them back together, this is probably suitable for a supplement, because we can't tease out the immune composition within the brain tumors. It is possible we could expand the CNS figure to look at subtypes of certain tumors we have subtypes for and see something - maybe worth exploring?

**Comment by komalsrathi** [comment]
@jharenza I am currently using `broad_histology`, are the subtypes in `pathology_diagnosis`?  Let me know, I will regenerate those figures and upload here for review.

**Comment by jharenza** [comment]
they are in `molecular_subtype` - we just have MB and high-grade gliomas/DMGs subtyped thus far

**Comment by komalsrathi** [comment]
@jharenza With molecular subtypes: I am using `broad_histology` where no subtype info, and `broad_histlogy-molecular_subtype` where subtype info present:

[heatmap_CIBERSORT(abs.).pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4307620/heatmap_CIBERSORT.abs.pdf)
[heatmap_xCell.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4307621/heatmap_xCell.pdf)
[corrplot_xCell_vs_CIBERSORT(abs.).pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4307622/corrplot_xCell_vs_CIBERSORT.abs.pdf)

**Comment by jaclyn-taroni** [comment]
@jharenza - what would you like to do here? Do you want to use 
> `broad_histology` where no subtype info, and` broad_histlogy-molecular_subtype` where subtype info present
If so, this code needs to be updated and reviewed before merging.

**Comment by jharenza** [comment]
I think this looks more informative with the subtypes, but also would make more sense for:
1. Medulloblastomas to be annotated as such, rather than `Embryonal tumor-subtype`
2. Embryonal tumor without subtypes to be annotated as the `ATRT`s they are.

I think these tweaks will be helpful for better expert and audience review.

**Comment by jaclyn-taroni** [comment]
@jharenza if you think that _most figures_ should be organized that way, I strongly encourage you to file a discussion ticket about that because it's going to be easier to start to address while we are working on #571.

**Comment by jharenza** [comment]
@jaclyn-taroni sure - I think it depends. For oncoprints, for ex, we could add subtype as a track. Let me create the discussion ticket!

**Comment by jharenza** [comment]
Discussion ticket here: https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/627

**Comment by komalsrathi** [comment]
@jharenza @jaclyn-taroni I have made these changes locally (reg. MB and ATRTs). Because the discussion is still open on the best way to annotate histologies, do you want me to wait until a decision is made and then push the changes?

**Comment by jaclyn-taroni** [comment]
@komalsrathi it comes down to your preference. Would you rather get this in as it stands and open another pull request later if things change or would you preference to wait and make that change in this PR?

**Comment by komalsrathi** [comment]
> @komalsrathi it comes down to your preference. Would you rather get this in as it stands and open another pull request later if things change or would your preference be to wait and make that change in this PR?

@jaclyn-taroni The first option sounds good to me.

File: analyses/immune-deconv/02-summary-plots.R [hunk 1]
```diff
@@ -29,6 +29,16 @@ deconvout <- opt$input
...
+# if short histology is Medulloblastoma or ATRT, use them as broad histology
+deconv.res$broad_histology <- ifelse(deconv.res$short_histology %in% c("ATRT", "Medulloblastoma"),
+                                     deconv.res$short_histology, deconv.res$broad_histology)
+
+# add molecular subtype info
+deconv.res$broad_histology <- ifelse(is.na(deconv.res$molecular_subtype), 
+                                     deconv.res$broad_histology, 
+                                     paste0(deconv.res$broad_histology, '-', deconv.res$molecular_subtype))
+
+
```
Comment (jaclyn-taroni) [inline analyses/immune-deconv/02-summary-plots.R]: My preference would be to create a new variable/column (perhaps called something like `disease_labels`) that uses the logic you have here but doesn't replace `broad_histology`. The new column could then be used downstream. I think this is less confusing than someone reading `broad_histology` in your heatmap code and having the end result in the plot using "custom" labels. You will have to revisit this anyway (related to https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/627), so I think that change can wait.

Comment (komalsrathi) [reply analyses/immune-deconv/02-summary-plots.R]: Hmm, makes sense. Thank you!

**Comment by komalsrathi** [comment]
@jaclyn-taroni Just checking in - was any decision made regarding the histology labels?

**Comment by jaclyn-taroni** [comment]
Hi @komalsrathi, no I think we are waiting for more folks from D3b with domain expertise to weigh in on #627 (+ trying to wrap up remaining subtyping tickets).

**Pull request merged**